client/src/components/DataFiles/DataFilesBreadcrumbs/DataFilesBreadcrumbs.jsx

```diff
@@ -119,16 +120,30 @@ const DataFilesBreadcrumbs = ({
   const paths = [];
   const pathComps = [];
 
+  const systems = useSelector(
+    (state) => state.systems.storage.configuration.filter((s) => !s.hidden),
+    shallowEqual
+  );
+
+  const root = useSystemDisplayName({ scheme, system });
+
+  const homeDir = systems.find((s) => s.name == root)?.homeDir;
+  const systemHomeDirPaths = homeDir?.split('/').filter((x) => !!x);
```

Comment:
In `useSystemDisplayName` we do use the system `name` and `scheme` to get a unique system but yes I agree it is pretty brittle. Ideally I'd like it if we had a `system_id` field for each system which would be a unique uuid but I am not sure how feasible it is to add that. Another potential solution could be that instead of passing around the system name, scheme or homeDir values we pass the entire system object (the system the user is currently viewing) to do the equals comparison with, this will effect performance a bit but will ensure uniqueness and won't be as brittle.

Reply:
System object, as in the Tapis definition?

Reply:
Created task [TV3-140](https://jira.tacc.utexas.edu/browse/TV3-140) where we can go into detail about a potential solution for this